processor/spanmetricsprocessor/processor.go

```diff
@@ -223,10 +236,49 @@ func validateRenames(renames []Rename) error {
 		if rename.NewCallsTotalMetricName == "" || rename.NewLatencyMetricName == "" {
 			return fmt.Errorf("renames: new metric name must be specified")
 		}
+
+		for _, attribute := range rename.Attributes {
```

Comment:
This is a bit inefficient. How about we have the single `Renames` struct like this:

```
type Rename struct {
	Attributes              []AttributeRenameMatchValues `mapstructure:"attributes"`
	NewCallsTotalMetricName string                       `mapstructure:"new_calls_total_metric_name"`
	NewLatencyMetricName    string                       `mapstructure:"new_latency_metric_name"`
}

type AttributeRenameMatchValues struct {
	Attribute           Dimension `mapstructure:"attribute"`
	AttributeValueRegex string    `mapstructure:"attribute_value_regex"`
        AttributeValueRegex *regexp.Regexp
}
```
And these methods:
````
func(r Rename) generateAttrKVRegex err {
  //  move the logic from `buildInternalRename()` to here
  ...
}

func(r Rename) valid bool {...}

```

Reply:
first change sounds good, I'll remodify the structure to just have the one.

For the function rewrite you did, I'm not sure I understand what's going on. The buildInternalRename should still be callled on startup so we don't need to keep recalling it. And I'm not sure why there's a valid bool? Or regexInitialised bool? We should do all this checking and initialisation on start up?